### PR TITLE
Geo projections

### PR DESCRIPTION
**Pull request by VeraZab**
opened an issue with plotly.js here: https://github.com/plotly/plotly.js/issues/2320
cc @nicolaskruchten

### CODE REVIEW

File: src/components/fields/Dropdown.js [hunk 1]
```diff
@@ -6,16 +6,31 @@ import {connectToContainer} from 'lib';
...
+      }
     }
 
     return (
       <Field {...this.props}>
         <DropdownWidget
           backgroundDark={this.props.backgroundDark}
-          options={this.props.options}
+          options={options}
           value={this.props.fullValue}
```
Comment (nicolaskruchten) [inline src/components/fields/Dropdown.js]: if you're pulling stuff out of `props` you may as well reuse them here :)

Comment (VeraZab) [reply src/components/fields/Dropdown.js]: oh forgot about this one : ) thanks!

Comment (nicolaskruchten) [reply src/components/fields/Dropdown.js]: np. For what it's worth, I'm not a huge fan of giant `const {...} = this.props;` everywhere... I'm much happier just having `this.props.x` everywhere as it's less mental indirection when reading the code. Basically if I see `this.props.x` I *know* it's from props and hasn't been locally mutated. If I see just `x` I have to look up higher in the code to know where it came from... Is it `const`? Was it `let`ed and mutated 9 times? etc... :)

File: src/components/fields/Dropdown.js [hunk 2]
```diff
@@ -11,11 +11,25 @@ export class UnconnectedDropdown extends Component {
       placeholder = this.props.fullValue;
     }
 
+    let options = this.props.options;
+
+    if (
+      this.props.attr === 'geo.projection.type' &&
```
Comment (nicolaskruchten) [inline src/components/fields/Dropdown.js]: Is there any way we can do this outside of the `Dropdown`? I'd rather not bake in this super-specific attribute stuff. Can we create a `ProjectionsDropdown` instead?

File: src/default_panels/GraphCreatePanel.js [hunk 3]
```diff
@@ -10,81 +10,122 @@ import {
...
+const GraphCreatePanel = ({localize: _}) => {
+  const geoProjectionOptions = {
+    usa: [
+      {
+        label: _('Albers USA'),
+        value: 'albers usa',
+      },
+    ],
+    other: [
+      {label: _('Equirectangular'), value: 'equirectangular'},
```
Comment (nicolaskruchten) [inline src/default_panels/GraphCreatePanel.js]: Same here, can we push this all into a `ProjectionsDropdown` and a `GeoScopeDropdown` ?

Comment (nicolaskruchten) [reply src/default_panels/GraphCreatePanel.js]: An additional advantage of this approach is that we could get rid of `LayoutSectionOverride` by just binding those two custom fields to the Layout.

File: src/components/index.js [hunk 4]
```diff
@@ -5,32 +5,34 @@ import {
   AxesRange,
   AxesSelector,
   CanvasSize,
-  ContourNumeric,
   ColorPicker,
+  ColorscalePicker,
+  ContourNumeric,
```
Comment (nicolaskruchten) [inline src/components/index.js]: Which ESLint setting are you using that reorders imports like this? I'll turn it on myself so we get less churn like this where your editor fixes my mistakes in your commits :)

File: src/components/containers/derived.js [hunk 5]
```diff
@@ -8,19 +8,12 @@ import {
...
-  supplyLayoutPlotProps,
 } from 'lib';
 
 const LayoutPanel = connectLayoutToPlot(Panel);
 
 const AxesFold = connectAxesToLayout(Fold);
 
-// Special purpose Section for usage when nested inside a Trace connected container.
-// supplyLayoutPlotProps overrides the trace context with a layout context.
-const LayoutSectionOverride = connectLayoutToPlot(Section, {
```
Comment (nicolaskruchten) [inline src/components/containers/derived.js]: yay!

File: src/components/fields/derived.js [hunk 6]
```diff
@@ -234,3 +235,93 @@ function computeAxesRefOptions(axes) {
...
+        {label: _('Stereographic'), value: 'stereographic'},
+        {label: _('Mollweide'), value: 'mollweide'},
+        {label: _('Hammer'), value: 'hammer'},
+        {label: _('Transverse Mercator'), value: 'transverseMercator'},
+        {label: _('Winkel Tripel'), value: 'winkel3'},
+        {label: _('Aitoff'), value: 'aitoff'},
+        {label: _('Sinusoidal'), value: 'sinusoidal'},
+      ];
+
+      if (
```
Comment (nicolaskruchten) [inline src/components/fields/derived.js]: oh yeah, this is so much nicer than in `Dropdown` right? :)

Comment (VeraZab) [reply src/components/fields/derived.js]: right :)

**Pull request merged**